src/Pipeline.cpp

```diff
@@ -658,8 +665,10 @@ void *Pipeline::compile_jit(const Target &target_arg) {
     // Compile to a module
     Module module = compile_to_module(args, name, target);
 
-    // Make sure we're not embedding any images
-    internal_assert(module.buffers().empty());
+    // We need to infer the arguments again, because compiling (GPU
+    // and offload targets) might have added new buffers we need to
+    // embed.
+    infer_arguments(module.functions().back().body);
```

Comment:
I was looking at this again to figure out how to implement it differently, and when looking at it, I noticed that this shouldn't increase the cost of realize.

Note that this PR mostly just changes the behavior of `compile_to_module`, `infer_arguments` still behaves as it used to when called in the same way (without a lowered body stmt).

Now note that in `compile_jit`, the call to `compile_to_module` (https://github.com/halide/Halide/blob/master/src/Pipeline.cpp#L659) is after an early-out if the pipeline is already compiled (https://github.com/halide/Halide/blob/master/src/Pipeline.cpp#L639).